h3-webtransport/src/server.rs

```diff
@@ -0,0 +1,422 @@
+
+        //= https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3/#section-3.3
+        let response = if validate_wt_connect(&request) {
+            Response::builder()
+                // This is the only header that chrome cares about.
+                .header("sec-webtransport-http3-draft", "draft02")
+                .status(StatusCode::OK)
+                .body(())
+                .unwrap()
+        } else {
+            Response::builder()
+                .status(StatusCode::BAD_REQUEST)
+                .body(())
+                .unwrap()
+        };
+
+        tracing::info!("Sending response: {response:?}");
+        stream.send_response(response).await?;
+
+        tracing::info!("Stream id: {}", stream.id());
+        let session_id = stream.send_id().into();
+        tracing::info!("Established new WebTransport session with id {session_id:?}");
+        let conn_inner = conn.inner.conn.lock().unwrap();
+        let opener = Mutex::new(conn_inner.opener());
+        drop(conn_inner);
+
+        Ok(Self {
+            session_id,
+            opener,
+            conn: Mutex::new(conn),
+            connect_stream: stream,
+        })
+    }
+
+    /// Receive a datagram from the client
+    pub fn accept_datagram(&self) -> ReadDatagram<C> {
+        ReadDatagram {
+            conn: self.conn.lock().unwrap().inner.conn.clone(),
+        }
+    }
+
+    /// Sends a datagram
+    ///
+    /// TODO: maybe make async. `quinn` does not require an async send
+    pub fn send_datagram(&self, data: impl Buf) -> Result<(), Error> {
+        self.conn
+            .lock()
+            .unwrap()
+            .send_datagram(self.connect_stream.id(), data)?;
+
+        Ok(())
+    }
+
+    /// Accept an incoming unidirectional stream from the client, it reads the stream until EOF.
+    pub fn accept_uni(&self) -> AcceptUni<C> {
+        AcceptUni { conn: &self.conn }
+    }
+
+    /// Accepts an incoming bidirectional stream or request
+    pub async fn accept_bi(&self) -> Result<Option<AcceptedBi<C>>, Error> {
+        // Get the next stream
+        // Accept the incoming stream
+        let stream = poll_fn(|cx| {
+            let mut conn = self.conn.lock().unwrap();
+            conn.poll_accept_request(cx)
+        })
+        .await;
+
+        tracing::debug!("Received biderectional stream");
+
+        let mut stream = match stream {
+            Ok(Some(s)) => FrameStream::new(BufRecvStream::new(s)),
+            Ok(None) => {
+                // We always send a last GoAway frame to the client, so it knows which was the last
+                // non-rejected request.
+                // self.shutdown(0).await?;
+                todo!("shutdown");
+                // return Ok(None);
+            }
+            Err(err) => {
+                match err.inner.kind {
+                    h3::error::Kind::Closed => return Ok(None),
+                    // h3::error::Kind::Application {
```

Comment:
@ten3roberts had to comment this out because I am getting: 

```
error[E0638]: `..` required with variant marked as non-exhaustive
   --> h3-webtransport/src/server.rs:177:21
    |
177 | /                     h3::error::Kind::Application {
178 | |                         code,
179 | |                         reason,
180 | |                         level: ErrorLevel::ConnectionError,
181 | |                     } => {
    | |_____________________^
```